### PR TITLE
TASK: Do not throw exception if no image is present

### PR DESCRIPTION
**Pull request by gerhard-boden**
**What I did**
`ImageViewHelper` (including the URI variant) will now recover if no image is present. 

**How I did it**
If the image argument is empty, the ViewHelper will now return empty.

**How to verify it**
Use the ViewHelper and and do not check if the image exists. No exception should be thrown.

**Checklist**
- [x] Code follows the PSR-2 coding style
- [x] Tests have been created, run and adjusted as needed
- [x] The PR is created against the [lowest maintained branch](https://www.neos.io/features/release-roadmap.html)

### CODE REVIEW

**Comment by gerhard-boden** [comment]
I decided not to declare this a bugfix since the old behavior was not by all means faulty. This change makes the ViewHelpers less sensitive for errors.

File: TYPO3.Media/Classes/TYPO3/Media/ViewHelpers/ImageViewHelper.php [hunk 1]
```diff
@@ -115,12 +115,16 @@ public function initializeArguments()
      * @param string $preset Preset used to determine image configuration
      * @return string an <img...> html tag
      */
-    public function render(ImageInterface $image = null, $width = null, $maximumWidth = null, $height = null, $maximumHeight = null, $allowCropping = false, $allowUpScaling = false, $async = false, $preset = null)
+    public function render($image = null, $width = null, $maximumWidth = null, $height = null, $maximumHeight = null, $allowCropping = false, $allowUpScaling = false, $async = false, $preset = null)
```
Comment (bwaidelich) [inline TYPO3.Media/Classes/TYPO3/Media/ViewHelpers/ImageViewHelper.php]: Why did you remove the type hint?

Comment (gerhard-boden) [inline TYPO3.Media/Classes/TYPO3/Media/ViewHelpers/ImageViewHelper.php]: you're right ofc, no need to remove them. Fixed!

Comment (gerhard-boden) [inline TYPO3.Media/Classes/TYPO3/Media/ViewHelpers/ImageViewHelper.php]: ah now i remember, I think i did it so empty strings could pass too. But we don't really need that because the fix is mainly for missing images which are ``null``.

File: TYPO3.Media/Tests/Unit/ViewHelpers/ImageViewHelperTest.php [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+{
+    /**
+     * var \TYPO3\Media\ViewHelpers\ImageViewHelper
+     */
+    protected $viewHelper;
+
+    public function setUp()
+    {
+        parent::setUp();
+        $this->viewHelper = new \TYPO3\Media\ViewHelpers\ImageViewHelper();
```
Comment (daniellienert) [inline TYPO3.Media/Tests/Unit/ViewHelpers/ImageViewHelperTest.php]: Please always use relative namespaces for more readability and to support the namespace change.

**Comment by daniellienert** [comment]
Thanks for the PR @gerhard-boden - as the image argument was not required, but though throwed an exception if not given, I would consider this as a Task / Bugfix although it changes the behavior.

**Comment by gerhard-boden** [comment]
@daniellienert thanks for the review, removed the absolute namespaces! Do you think we should retarget this to 2.1?

**Comment by daniellienert** [comment]
@gerhard-boden - No I would leave it targeted at master as it is.

**Comment by kitsunet** [comment]
If you rebase this to current master, the failing test should be gone ;)

**Comment by gerhard-boden** [comment]
rebased and fixed broken tests to work again after fluid merge

**Comment by daniellienert** [comment]
If no one objects, I would like to merge it.

File: TYPO3.Media/Classes/TYPO3/Media/ViewHelpers/Uri/ImageViewHelper.php [hunk 3]
```diff
@@ -68,7 +68,7 @@ public function initializeArguments()
     /**
      * Renders the path to a thumbnail image, created from a given image.
      *
-     * @param ImageInterface $image
+     * @param ImageInterface|null $image
```
Comment (mficzel) [inline TYPO3.Media/Classes/TYPO3/Media/ViewHelpers/Uri/ImageViewHelper.php]: In my test flow did not understand  the ``|null`` properly. If i change this line the change works as expected.

File: TYPO3.Media/Classes/TYPO3/Media/ViewHelpers/ImageViewHelper.php [hunk 4]
```diff
@@ -104,7 +104,7 @@ public function initializeArguments()
     /**
      * Renders an HTML img tag with a thumbnail image, created from a given image.
      *
-     * @param ImageInterface $image The image to be rendered as an image
+     * @param ImageInterface|null $image The image to be rendered as an image
```
Comment (mficzel) [inline TYPO3.Media/Classes/TYPO3/Media/ViewHelpers/ImageViewHelper.php]: In my test flow did not understand  the ``|null`` properly. If i change this line the change works as expected.

Comment (gerhard-boden) [inline TYPO3.Media/Classes/TYPO3/Media/ViewHelpers/ImageViewHelper.php]: Just tried it again, can't reproduce. I could take out the `|null` fallback, but would still be interesting to know why it works on my system and not on yours. Maybe we can compare later or tommorrow.

Comment (gerhard-boden) [inline TYPO3.Media/Classes/TYPO3/Media/ViewHelpers/ImageViewHelper.php]: Update: Now it failed when the image was empty ofc, changed the type to mixed (because I noticed that the image can also be an empty string after being removed from within the backend).

regression:
https://github.com/neos/neos-development-collection/pull/1203/commits/8cde724a22ed65b8d4d5ac4cc2ee9290f429170c

**Comment by mficzel** [comment]
@daniellienert can you check this again since this was adjusted since your review. I'd like to merge this soonish.

**Comment by mficzel** [comment]
@daniellienert can you check this again. Id like to merge this but the code was changed since your review.

**Comment by mficzel** [comment]
Note: anyone else is welcome to approve this aswell ... we just need one more approval

**Comment by johannessteu** [comment]
@skurfuerst and me reviewed this; full ack this is fine as-is.
